app/controllers/TransferCompleteController.scala

```diff
@@ -14,17 +15,29 @@ import scala.concurrent.ExecutionContext
 class TransferCompleteController @Inject() (
     val controllerComponents: SecurityComponents,
     val keycloakConfiguration: KeycloakConfiguration,
-    val consignmentService: ConsignmentService
+    val consignmentService: ConsignmentService,
+    val messagingService: MessagingService
 )(implicit val ec: ExecutionContext)
     extends TokenSecurity
     with I18nSupport {
 
   def transferComplete(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
-    consignmentService
-      .getConsignmentRef(consignmentId, request.token.bearerAccessToken)
-      .map { consignmentReference =>
-        Ok(views.html.standard.transferComplete(consignmentId, consignmentReference, request.token.name))
-      }
+    for {
+      consignmentTransferSummary <- consignmentService.getConsignmentConfirmTransfer(consignmentId, request.token.bearerAccessToken)
+      consignmentReference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+    } yield {
+      messagingService.sendTransferCompleteNotification(
+        TransferCompleteEvent(
+          transferringBodyName = consignmentTransferSummary.transferringBodyName,
+          consignmentReference = consignmentReference,
```

Comment:
You can use `consignmentTransferSummary.consignmentReference` so you don't need to make an extra call to get the consignment reference.